src/ex2/ItemManager.js

```diff
@@ -0,0 +1,170 @@
+        let promise = null;
+        if (!pokemonInList) {
+          promise = this.fetchPokemonAndAddToList(pokemonNumber);
+        }
+        promises.push(promise);
+      } else {
+        //task is not a number
+        this.handleAddTodo(item);
+      }
+    }
+
+    Promise.all(promises).then((data) => {
+      //console.log("Values", data);
+      // console.log(this.pokemonErrorBatch)
+      if (!this.pokemonErrorBatch.length) {
+        return;
+      }
+      if (this.pokemonErrorBatch.length == 1) {
+        this.handleAddTodo(
+          `Pokemon with ID ${this.pokemonErrorBatch[0]} was not found`
+        );
+        this.pokemonErrorBatch = [];
+      }
+
+      if (this.pokemonErrorBatch.length > 1) {
+        const errorsToString = this.pokemonErrorBatch.join();
+        this.handleAddTodo(
+          `Failed to fetch pokemon with this input: ${errorsToString}`
+        );
+      }
+    });
+
+    this.pokemonErrorBatch = [];
+  }
+
+  handleAddTodo(itemTextValue) {
+    /*   const itemInList = this.itemList.find(function (item, index) {
+      if (item.name == itemTextValue) {
+       
+        return true;
+      }
+    });
+
+    if (itemInList) {
+      return;
+    }  */
+
+    this.addTodo(itemTextValue);
+
+    const getTrashButtonIdToHandleRemove = () => {
+     // alert(counter)
+      const trashButton = document.getElementById(`${counter}TrashID`);
+      trashButton.onclick = (event) => {
+        
+        console.log(event.currentTarget.id);
+        const itemToRemove = trashButton.parentElement
+        this.handleRemoveTodo(itemToRemove);
+      };
+    };
+
+    //need to add also save to local storage
+    let returned;
+    const counter = this.uiManager.idCounter;
+    if (this.itemList.length == 1) {
+      this.uiManager.showButtonsAndFooter();
+
+      this.uiManager.UIHandleAddRenderItem(itemTextValue, this.itemList.length);
+
+      getTrashButtonIdToHandleRemove(counter);
+      return;
+    }
+    this.uiManager.UIHandleAddRenderItem(itemTextValue, this.itemList.length);
+
+    getTrashButtonIdToHandleRemove(counter);
+  }
+
+  addTodo(itemTextValue) {
+    this.item = new Item(itemTextValue);
+    this.itemList.push(this.item);
+  }
+
+  handleRemoveTodo(itemToRemove) {
+    const itemTextValueToRemove = itemToRemove.firstChild.textContent;
+    this.removeTodoFromItemList(itemTextValueToRemove);
+    this.uiManager.UIremoveItem(itemToRemove, this.itemList.length);
+  }
+
+  removeTodoFromItemList(itemToRemove) {
+    this.itemList = this.itemList.filter((item) => item.name != itemToRemove);
+  }
+
+  handleClearAllTodos() {
+    this.itemList = [];
+    this.fethcedPokemons = [];
+    this.uiManager.UIclearAllTodos();
+    this.uiManager.hideButtonsAndFooter();
+  }
+
+  async fetchPokemonAndAddToList(pokemonNumber) {
+    const pokemon = await this.pokemonClient.fetchPokemon(pokemonNumber);
+
+    //catch ${pokemonName} the ${pokemonType} pokemon
+    if (pokemon != `Pokemon was not found`) {
```

Comment:
since your "if" is longer then what's after, it would be better practice to reverse the conditions and so something like 
```
if (pokemon == "not found") {
 add_error
 return
}
... rest of code
```
Also, I don't recommend comparing to strings in other parts of the code. It's better to use constants so if you change it in one file it won't break your flow.